### PR TITLE
Add stages for springlock

### PR DESCRIPTION
**Pull request by nicholast013**
unknown

## About The Pull Request

Stages springlock is great than old one it add those feeling of being slowly springlocked, I have fixed the messy code thank you for people that helping me

## Why It's Good For The Game

Old springlock boring its just snap and killed you, how about stages springlock! Look as someone scream while slowly modsuit exoskeleton insert into their body

## Changelog

:cl:
add: stages to springlock and modsuit cant be remove once springlocked
/:cl:

### CODE REVIEW

File: code/modules/mod/modules/modules_maint.dm [hunk 1]
```diff
@@ -48,12 +48,60 @@
 		return
 	mod.wearer.visible_message("[src] inside [mod.wearer]'s [mod.name] snaps shut, mutilating the user inside!", span_userdanger("*SNAP*"))
 	mod.wearer.emote("scream")
+	mod.wearer.Paralyze(10000)
```
Comment (ATH1909) [inline code/modules/mod/modules/modules_maint.dm:51]: hulk momento

File: code/modules/mod/modules/modules_maint.dm [hunk 2]
```diff
@@ -48,12 +48,60 @@
...
+
+///Stage 6
+/obj/item/mod/module/springlock/proc/stage_6()
+	mod.wearer.apply_damage(damage = 45, damagetype = BRUTE, def_zone = BODY_ZONE_HEAD , forced = TRUE, spread_damage = TRUE, sharpness = WOUND_BLUNT)
+	playsound(mod.wearer, 'sound/effects/snap.ogg', 75, TRUE, frequency = 0.5)
+	playsound(mod.wearer, 'sound/effects/splat.ogg', 50, TRUE, frequency = 0.5)
+	mod.wearer.bleed(8)
 	if(!HAS_TRAIT(mod.wearer, TRAIT_NODEATH))
-		mod.wearer.death() //just in case, for some reason, they're still alive
+		mod.wearer.death()
```
[the remark this answers is not in the document]
Comment (Iamgoofball) [reply code/modules/mod/modules/modules_maint.dm:104]: it's supposed to kill you

File: code/modules/mod/modules/modules_maint.dm (hunk 1 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:51]: alright

File: code/modules/mod/modules/modules_maint.dm [hunk 3]
```diff
@@ -33,12 +33,12 @@
 	playsound(src, 'sound/items/modsuit/springlock.ogg', 75, TRUE)
 	addtimer(CALLBACK(src, .proc/snap_shut), rand(3 SECONDS, 5 SECONDS))
 	RegisterSignal(mod, COMSIG_MOD_ACTIVATE, .proc/on_activate_spring_block)
+	UnregisterSignal(mod.wearer, COMSIG_ATOM_EXPOSE_REAGENTS)
 
 ///Signal fired when wearer attempts to activate/deactivate suits
 /obj/item/mod/module/springlock/proc/on_activate_spring_block(datum/source, user)
 	SIGNAL_HANDLER
 
-	balloon_alert(user, "springlocks aren't responding...?")
```
Comment (tralezab) [inline code/modules/mod/modules/modules_maint.dm:41]: justify this?

File: code/modules/mod/modules/modules_maint.dm [hunk 4]
```diff
@@ -48,12 +48,60 @@
...
 	mod.wearer.visible_message("[src] inside [mod.wearer]'s [mod.name] snaps shut, mutilating the user inside!", span_userdanger("*SNAP*"))
 	mod.wearer.emote("scream")
+	mod.wearer.Paralyze(10000, TRUE)
 	playsound(mod.wearer, 'sound/effects/snap.ogg', 75, TRUE, frequency = 0.5)
 	playsound(mod.wearer, 'sound/effects/splat.ogg', 50, TRUE, frequency = 0.5)
 	mod.wearer.client?.give_award(/datum/award/achievement/misc/springlock, mod.wearer)
-	mod.wearer.apply_damage(500, BRUTE, forced = TRUE, spread_damage = TRUE, sharpness = SHARP_POINTY) //boggers, bogchamp, etc
+	mod.wearer.apply_damage(damage = 45, damagetype = BRUTE, def_zone = BODY_ZONE_R_ARM, forced = FALSE, spread_damage = FALSE, sharpness = WOUND_BLUNT)
+	ADD_TRAIT(mod, TRAIT_NODROP, HAND_REPLACEMENT_TRAIT)
+	addtimer(CALLBACK(src, .proc/stage_2), rand(8 SECONDS, 10 SECONDS))
```
Comment (tralezab) [inline code/modules/mod/modules/modules_maint.dm:57]: 8 to 10 seconds 6 times can lead to up to a full minute of dying, which obviously is ridiculous. Am I alone in that?

**Review by tralezab — CHANGES_REQUESTED** [review]
just some things i noticed

File: code/modules/mod/modules/modules_maint.dm (hunk 4 above)
Comment (IndieanaJones) [reply code/modules/mod/modules/modules_maint.dm:57]: I mean, it's a slow and painful death.  You're already committing yourself to a very flashy and extravagant death, might as well make it last.

File: code/modules/mod/modules/modules_maint.dm (hunk 3 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:41]: i just remove it cause i dont know what it does

File: code/modules/mod/modules/modules_maint.dm (hunk 4 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:57]: very slow and painful death is the objective

File: code/modules/mod/modules/modules_maint.dm (hunk 3 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:41]: and its just gonna alert user nothing important

Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:41]: justify this?

File: code/modules/mod/modules/modules_maint.dm [hunk 5]
```diff
@@ -48,12 +48,60 @@
 		return
 	mod.wearer.visible_message("[src] inside [mod.wearer]'s [mod.name] snaps shut, mutilating the user inside!", span_userdanger("*SNAP*"))
 	mod.wearer.emote("scream")
+	mod.wearer.Paralyze(10000, TRUE)
 	playsound(mod.wearer, 'sound/effects/snap.ogg', 75, TRUE, frequency = 0.5)
 	playsound(mod.wearer, 'sound/effects/splat.ogg', 50, TRUE, frequency = 0.5)
 	mod.wearer.client?.give_award(/datum/award/achievement/misc/springlock, mod.wearer)
-	mod.wearer.apply_damage(500, BRUTE, forced = TRUE, spread_damage = TRUE, sharpness = SHARP_POINTY) //boggers, bogchamp, etc
+	mod.wearer.apply_damage(damage = 45, damagetype = BRUTE, def_zone = BODY_ZONE_R_ARM, forced = FALSE, spread_damage = FALSE, sharpness = WOUND_BLUNT)
+	ADD_TRAIT(mod, TRAIT_NODROP, HAND_REPLACEMENT_TRAIT)
```
Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:56]: ...why?

File: code/modules/mod/modules/modules_maint.dm [hunk 6]
```diff
@@ -48,12 +48,60 @@
...
+	addtimer(CALLBACK(src, .proc/stage_2), rand(8 SECONDS, 10 SECONDS))
+
+///Stage 2
+/obj/item/mod/module/springlock/proc/stage_2()
+	mod.wearer.emote("scream")
+	mod.wearer.bleed(8)
+	playsound(mod.wearer, 'sound/effects/snap.ogg', 75, TRUE, frequency = 0.5)
+	playsound(mod.wearer, 'sound/effects/splat.ogg', 50, TRUE, frequency = 0.5)
+	mod.wearer.apply_damage(damage = 40, damagetype = BRUTE, def_zone = BODY_ZONE_R_LEG, forced = FALSE, spread_damage = FALSE, sharpness = WOUND_BLUNT)
+	addtimer(CALLBACK(src, .proc/stage_3), rand(8 SECONDS, 10 SECONDS))
```
Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:66]: all of this is just copypaste, why not make it a proc that you pass an argument of def_zone to

File: code/modules/mod/modules/modules_maint.dm [hunk 7]
```diff
@@ -48,12 +48,60 @@
...
+	mod.wearer.emote("scream")
+	mod.wearer.bleed(8)
+	playsound(mod.wearer, 'sound/effects/snap.ogg', 75, TRUE, frequency = 0.5)
+	playsound(mod.wearer, 'sound/effects/splat.ogg', 50, TRUE, frequency = 0.5)
+	mod.wearer.apply_damage(damage = 50, damagetype = BRUTE, def_zone = BODY_ZONE_CHEST, forced = FALSE, spread_damage = FALSE, sharpness = WOUND_BLUNT)
+	addtimer(CALLBACK(src, .proc/stage_6), rand(8 SECONDS, 10 SECONDS))
+
+///Stage 6
+/obj/item/mod/module/springlock/proc/stage_6()
+	mod.wearer.apply_damage(damage = 45, damagetype = BRUTE, def_zone = BODY_ZONE_HEAD , forced = TRUE, spread_damage = TRUE, sharpness = WOUND_BLUNT)
```
Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:99]: why is it randomly forced, spread damage

File: code/modules/mod/modules/modules_maint.dm [hunk 8]
```diff
@@ -33,12 +33,12 @@
 	playsound(src, 'sound/items/modsuit/springlock.ogg', 75, TRUE)
 	addtimer(CALLBACK(src, .proc/snap_shut), rand(3 SECONDS, 5 SECONDS))
 	RegisterSignal(mod, COMSIG_MOD_ACTIVATE, .proc/on_activate_spring_block)
+	UnregisterSignal(mod.wearer, COMSIG_ATOM_EXPOSE_REAGENTS)
```
Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:36]: what if someone is turning off their suit while this happens? wouldnt it runtime from 2 unregister signals of the same type?

File: code/modules/mod/modules/modules_maint.dm [hunk 9]
```diff
@@ -48,12 +48,60 @@
 		return
 	mod.wearer.visible_message("[src] inside [mod.wearer]'s [mod.name] snaps shut, mutilating the user inside!", span_userdanger("*SNAP*"))
 	mod.wearer.emote("scream")
+	mod.wearer.Paralyze(10000, TRUE)
```
Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:51]: are we sure during revival this doesnt keep going

File: code/modules/mod/modules/modules_maint.dm [hunk 10]
```diff
@@ -48,12 +48,60 @@
...
 	playsound(mod.wearer, 'sound/effects/snap.ogg', 75, TRUE, frequency = 0.5)
 	playsound(mod.wearer, 'sound/effects/splat.ogg', 50, TRUE, frequency = 0.5)
 	mod.wearer.client?.give_award(/datum/award/achievement/misc/springlock, mod.wearer)
```
Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:54]: this should be done at the end when the wearer dies

File: code/modules/mod/modules/modules_maint.dm (hunk 4 above)
Comment (Fikou) [inline code/modules/mod/modules/modules_maint.dm:57]: a full minute of dying is way too much, especially when the wearer is not even muted, make it crush their lungs and tongue or something or else all the viability of using it to kill someone is kinda ruined as they have a whole minute to call you out

Comment (Fikou) [reply code/modules/mod/modules/modules_maint.dm:57]: and do shorten it, 3-5 seconds is fine.

File: code/modules/mod/modules/modules_maint.dm (hunk 9 above)
Comment (Fikou) [reply code/modules/mod/modules/modules_maint.dm:51]: TRUE doesnt tell someone reading the code anything, use the named parameter

File: code/modules/mod/modules/modules_maint.dm (hunk 8 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:36]: Thats the problem

File: code/modules/mod/modules/modules_maint.dm (hunk 7 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:99]: I forgot

File: code/modules/mod/modules/modules_maint.dm (hunk 3 above)
Comment (Mothblocks) [reply code/modules/mod/modules/modules_maint.dm:41]: why DID you resolve the last one

File: code/modules/mod/modules/modules_maint.dm (hunk 4 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:57]: Alright

File: code/modules/mod/modules/modules_maint.dm (hunk 9 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:51]: Alright

File: code/modules/mod/modules/modules_maint.dm (hunk 8 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:36]: i need to solve that

File: code/modules/mod/modules/modules_maint.dm (hunk 6 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm:66]: Alright

File: code/modules/mod/modules/modules_maint.dm (hunk 3 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm]: i have readded it again

File: code/modules/mod/modules/modules_maint.dm (hunk 4 above)
Comment (nicholast013) [reply code/modules/mod/modules/modules_maint.dm]: time between stage now shorten to 3 or 5

**Comment by optimumtact** [comment]
we really don't need to expand on this meme.

**Pull request closed without merging**